dist/jquery.flot.js

```diff
@@ -3898,34 +3898,28 @@ Licensed under the MIT license.
         }
 
         function drawSeriesPoints(series, ctx, plotOffset, plotWidth, plotHeight, drawSymbol, getColorOrGradient) {
-            function plotPoints(datapoints, radius, fillStyle, offset, shadow, axisx, axisy, symbol) {
+            function drawCircle(ctx, x, y, radius, shadow, fill) {
+                ctx.moveTo(x + radius, y);
+                ctx.arc(x, y, radius, 0, shadow ? Math.PI : Math.PI * 2, false);
+            }
+            function plotPoints(datapoints, radius, fill, offset, shadow, axisx, axisy, drawSymbolFn) {
                 var points = datapoints.points,
                     ps = datapoints.pointsize;
 
+                ctx.beginPath();
                 for (var i = 0; i < points.length; i += ps) {
                     var x = points[i],
                         y = points[i + 1];
                     if (x == null || x < axisx.min || x > axisx.max || y < axisy.min || y > axisy.max) {
                         continue;
                     }
 
-                    ctx.beginPath();
                     x = axisx.p2c(x);
                     y = axisy.p2c(y) + offset;
 
-                    if (symbol === 'circle') {
-                        ctx.arc(x, y, radius, 0, shadow ? Math.PI : Math.PI * 2, false);
-                    } else if (typeof symbol === 'string' && drawSymbol && drawSymbol[symbol]) {
-                        drawSymbol[symbol](ctx, x, y, radius, shadow);
-                    }
-                    ctx.closePath();
-
-                    if (fillStyle) {
-                        ctx.fillStyle = fillStyle;
```

Comment:
Removing the fillStyle set outside the for loop made it another 5% faster but the GPU load was drastically reduced.